### ISSUE TITLE
[BUG] Support for python3 in logstash_mod

### ISSUE THREAD

**Issue by aokiji** [saltstack/salt#58001]
**Description**
Logstash module fails with python3

**Steps to Reproduce the behavior**
```
$ git clone https://github.com/aokiji/saltstack-logstash && cd saltstack-logstash
$ docker-compose build saltstack-py3
$ docker-compose run --rm saltstack-py3
Starting saltstack-logstash_logstash_1 ... done
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/handlers.py", line 622, in emit
    self.send(s)
  File "/usr/lib/python3.5/logging/handlers.py", line 680, in send
    self.sock.sendto(s, self.address)
TypeError: a bytes-like object is required, not 'str'
Call stack:
  File "/usr/bin/salt-run", line 10, in <module>
    salt_run()
  File "/usr/lib/python3/dist-packages/salt/scripts.py", line 443, in salt_run
    client.run()
  File "/usr/lib/python3/dist-packages/salt/cli/run.py", line 40, in run
    ret = runner.run()
  File "/usr/lib/python3/dist-packages/salt/runner.py", line 271, in run
    daemonize=False)
  File "/usr/lib/python3/dist-packages/salt/client/mixins.py", line 477, in _proc_function
    return self.low(fun, low, full_return=False)
  File "/usr/lib/python3/dist-packages/salt/client/mixins.py", line 377, in low
    data['return'] = func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/salt/runners/salt.py", line 106, in cmd
    if fun in functions \
  File "/usr/lib/python3/dist-packages/salt/modules/logmod.py", line 65, in error
    log.error(message)
  File "/usr/lib/python3.5/logging/__init__.py", line 1309, in error
    self._log(ERROR, msg, args, **kwargs)
  File "/usr/lib/python3/dist-packages/salt/log/setup.py", line 345, in _log
    self, level, msg, args, exc_info=exc_info, extra=extra
Message: 'executing 2019.2@py3'
Arguments: ()
[ERROR   ] executing 2019.2@py3
True
Couldnt log message
```

**Expected behavior**
That message is sent to logstash and can be viewed in logstash docker output

```
$ docker-compose logs logstash | grep '2019.2@py3'
```

**Comment by aokiji** [comment]
I have checked that this is solved in the latest version, if it could be backported it would be nice

**Comment by krionbsd** [comment]
@aokiji could you please provide `salt --versions-report`

**Comment by aokiji** [comment]
> @aokiji could you please provide `salt --versions-report`

```
docker-compose run --rm saltstack-py3 salt --versions-report
Starting saltstack-logstash_logstash_1 ... done
Salt Version:
           Salt: 2019.2.5

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.3 (default, Jul  9 2020, 13:00:10)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1

System Versions:
           dist: debian 9.13
         locale: ANSI_X3.4-1968
        machine: x86_64
        release: 5.6.13-100.fc30.x86_64
         system: Linux
        version: debian 9.13
```

**Comment by krionbsd** [comment]
Thanks @aokiji, we can't backport this fix to 2019.2.x branch as it's at stage 3 at the moment, only critical CVEs might be merged - [support cycle](https://www.saltstack.com/product-support-lifecycle/)
If you could upgrade to 3001, that would be awesome.

**Comment by sagetherage** [comment]
@aokiji closing this for now, mention me if you need anything here, thank you!
